File: swap-numbers/java/SwapNumbers.java
import java.util.Scanner;

public class SwapNumbers {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter first number: ");
        var firstNumber = scanner.nextInt();

        System.out.print("Enter second number: ");
        var secondNumber = scanner.nextInt();

        swapNumbersUsingThirdVariable(firstNumber, secondNumber);
        swapNumbersWithoutUsingThirdVariable(firstNumber, secondNumber);

        scanner.close();
    }

    static void swapNumbersUsingThirdVariable(int firstNumber, int secondNumber) {
        System.out.println("Swapping numbers using a third variable");

        var temp = firstNumber;
        firstNumber = secondNumber;
        secondNumber = temp;

        System.out.println("First number is:" + firstNumber);
        System.out.println("Second number is:" + secondNumber);
    }

    static void swapNumbersWithoutUsingThirdVariable(int firstNumber, int secondNumber) {
        System.out.println("Swapping numbers without using a third variable");

        firstNumber = firstNumber + secondNumber;
        secondNumber = firstNumber - secondNumber;
        firstNumber = firstNumber - secondNumber;

        System.out.println("First number is:" + firstNumber);
        System.out.println("Second number is:" + secondNumber);
    }

}

File: swap-numbers/python/SwapNumbers.py
def main():
    first_number = int(input('Enter first number: '))
    second_number = int(input('Enter second number: '))
    swap_numbers_using_third_number(first_number, second_number)
    swap_numbers_without_using_third_number(first_number, second_number)

def swap_numbers_using_third_number(first_number, second_number):
    print('Swapping numbers using a third variable...')
    
    temp = first_number
    first_number = second_number
    second_number = temp

    print(f'First number is: {first_number}')
    print(f'Second number is: {second_number}')

def swap_numbers_without_using_third_number(first_number, second_number):
    print('Swapping numbers without using a third variable...')
    
    first_number = first_number + second_number
    second_number = first_number - second_number
    first_number = first_number - second_number

    print(f'First number is: {first_number}')
    print(f'Second number is: {second_number}')
    
if __name__ == "__main__":
    main()